doc/ins_sign_stake_pool_registration.md

```diff
@@ -0,0 +1,191 @@
+# Stake Pool Registration
+
+## Description
+
+Stake pool registration certificates differ from other types of certificates in three respects:
+1. The certificate data does not fit a single APDU message.
+2. Transactions containing them are severely restricted due to security concerns about intermingling of witnesses (for instance, the same keys are used for witnessing a pool owner and a withdrawal).
+3. Transactions containing them potentially need to be signed by several unrelated sets of keys, often managed by separate entities.
+
+We therefore split the process into two separate workflows according to the so-called use cases describing who is signing the transaction (operator of the pool or one of its owners). The chosen usecase is given in the initial message for transaction signing (see [Stake Pool Registration](ins_sign_stake_pool_registration.md)). Note that the serialized transaction must be the same for both flows, only the required witnesses and some user interactions differ.
+
+The key derivation scheme for pool cold keys used by a pool operator is described in [CIP 1853 - HD Stake Pool Cold Keys for Cardano](https://cips.cardano.org/cips/cip1853/).
+
+## Use case for pool operator
+
+- Set by using `SIGN_TX_USECASE_POOL_REGISTRATION_OPERATOR=0x05` in the initial sign transaction APDU message (see [SignTx call](ins_sign_tx.md)).
+
+- The pool registration transaction is paid for by the operator which is thus supposed to provide witnesses for inputs. An additional witness corresponds to pool id (signed by pool cold key).
+
+- Pool id must be given by cold key derivation path.
+
+- Pool owners must be given by hash (of the staking key).
+
+## Use case for pool owner
+
+- Set by using `SIGN_TX_USECASE_POOL_REGISTRATION_OWNER=0x04` in the initial sign transaction APDU message (see [SignTx call](ins_sign_tx.md)).
+
+- Transaction outputs must be given by addresses (not address params) and are not shown to the user (because he is not funding the transaction). The fee is not shown either.
+
+- Only a single witness is allowed (the key derivation path is the standard staking key path of the owner). 
+
+- It is necessary to include exactly one pool owner given by path in the certificate; all the other owners must be given by their respective key hashes.
+
+- Pool id must be given by key hash.
+
+- Pool reward account must be given in binary, not as a derivation path.
```

Comment:
I think this requirement is superfluous - I see no harm in letting some of the owners own the rewards account